### DISCUSSION TITLE
Guideline of marker (Possible z-axis or anchor top?)

### DISCUSSION THREAD

**Discussion by ivanleong** [patrykandpatrick/vico#794] [Questions] [answered]
### Question

Is it possible to have the vertical guidelines of a marker to sit behind the bar columns instead of the front? I need to have the illusion that the guideline is either from behind or sits right on top of each columns like below:

![Screenshot 2024-07-12 at 10 41 08](https://github.com/user-attachments/assets/1cdfb366-b068-4dcb-b859-a94e3dcb9861)

### Vico version(s)

2.0.0-alpha.21

### UI framework(s)

Jetpack Compose

**Comment by ivanleong** [comment]
Any update on this? Is this not possible?

**Comment by Gowsky** [comment]
Yes, it's possible. You'd need to use a custom `CartesianMarker` implementation, or a subclass of `DefaultCartesianMarker`. The bottom edge of the guideline can be at [`canvasY`](https://patrykandpatrick.com/vico/api/2.0.0-alpha.27/core/com.patrykandpatrick.vico.core.cartesian.marker/-column-cartesian-layer-marker-target/-column/canvas-y). We also plan on adding a function to allow for markers to draw behind the chart content.

**Comment by Gowsky** [comment, marked as answer]
[Vico `2.0.0-beta.3`](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-beta.3) adds the feature I mentioned above. `CartesianMarker` now has two drawing functions, `drawUnderLayers` and `drawOverLayers`.
